lib/firmware/cosmos.c

```diff
@@ -0,0 +1,82 @@
+#include "keepkey/firmware/cosmos.h"
+
+#include "keepkey/board/confirm_sm.h"
+#include "keepkey/board/util.h"
+#include "keepkey/firmware/home_sm.h"
+#include "keepkey/firmware/storage.h"
+#include "trezor/crypto/segwit_addr.h"
+#include "trezor/crypto/ecdsa.h"
+#include <stdbool.h>
+#include <time.h>
+
+/* This was inlined from the trezor firmware because it was not exported.
```

Comment:
Just make a PR to drop `static` from github.com/keepkey/trezor-firmware. We can live with a few small downstream changes here and there.

Reply:
Should this change be made to master?

Reply:
The PR should be against the `keepkey` branch. `master` reflects upstream.

Reply:
I'm running into the issue now where it's upset about a missing ed25519-blake2b.c source file in the CMakeLists.txt at this point, so somewhere between the locked commit, and keepkey/master, this file was removed, what should we do here? can we remove that file from the CMakeLists? I think it's out of scope to fix build issues unrelated to the cosmos update.

Reply:
OK, so I tried just removing it but this is a thread that is unraveling a lot right now.

Reply:
I'll have a look.

Reply:
Can you try with this commit? https://github.com/keepkey/trezor-firmware/pull/1/files

Reply:
> Clearly keepkey/trezor-firmware/master has a lot of incompatible changes with the current branch of keepkey/keepkey-firmware/master. Please advise

`keepkey-firmware` is using `keepkey/trezor-firmware/keepkey` not` keepkey/trezor-firmware/master`, hence why I asked you to make the commit against the `keepkey` branch, not `master`.

Reply:
🤦‍♂ my bad. Thanks.

Reply:
:)

Reply:
was `mnemonic_wordlist` removed in this fast forward? I'm getting complaints about implicit declarations and I don't recall changing any of that code

Reply:
bee9b60e4ce0a48d62494148fd62344491e3b2a6 deals with that. upstream renamed it to `wordlist`, IIRC.

Reply:
you probably just need to merge latest `keepkey-firmware/master` into your branch.